bhohnco/TheWormhole
# Feature/location data display refactoring

### Is this a fix or a feature?
- Feature

### What is the change or fix?
- Refactored _top-artists.scss_ styling so that we now have a "artist card" grid display instead of a list of names.
- Refactored _TopArtists.js_ functionality so that we now create and display X number of "artist card" elements based on the number included in our `filteredArtists()` function.
- Refactored `apiCalls.getArtistImage()` function so that it now dynamically fetches a MovieBrainz artist object dynamically based on whatever ID is passed in.
- Built a `retrieveArtistImage()` function (_App.js_) to invoke the API call and set state with the returned data.  We night not actually need this, since the image property will just get assigned to each element in the `this.state.topArtists` array.  Could just make the API call right in _TopArtists.js_, with that in mind.  Probably something to figure out when we get into state management with Redux.  

### How should this be tested?
- Display in browser.  But hold off on running the commented-out code block until we've figure out the CORS issue!

EDIT: Also added / pushed up new "random country" functionality: now when you load or refresh the page, a different country will get assigned to `this.state.location`, and will be used to make TopArtists and TopTracks fetch requests.  I did this using a new `countries.js` dataset I put in the utilities directory, and using a few new functions in _utils.js_ and _App.js_.  Not all countries in the array have Last.fm data, so if the page ever fails to load, please check the console to see which country it was trying to fetch (should appear within an error or log), and then comment-out that country name in the _countries.js_ dataset.  I caught a handful of them, but any more page refreshes and I would've induced a seizure.  Let's just cut them out one at a time.
